### PR TITLE
Add: Login With Steam

### PR DESCRIPTION
**Pull request by TheDedemon**
Hi, the feature can be tested on my branch **tests/login_with_steam**. I didn't put it with the feature to avoid importing external extension in the main branch.

### CODE REVIEW

**Comment by TheDedemon** [comment]
I created an Issue corresponding to the PR, I forgot to add an example for the Steam Auth Ticket For Web API.

https://github.com/Structed/godot-playfab/issues/133

**Review by Structed — CHANGES_REQUESTED** [review]
A few bits of nitpicking, but overall really nice, thank you!

**Comment by Structed** [comment]
I feel like this and the actual integration (with the code you have in the docs) should be it's own `godot-playfab-steam` addon. What are your thoughts on this, @TheDedemon?

File: docs/user/Logins/login-steam.md [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+The [PlayFab](/addons/godot-playfab/PlayFab.gd) Node class implements the login with Steam using the method below.
+
+```gdscript
+    func login_with_steam(steam_auth_ticket: String, is_auth_ticket_for_api: bool, create_account: bool, info_request_parameters: GetPlayerCombinedInfoRequestParams) -> void:
+```
+
+- **steam_auth_ticket**: String that contains the steam authentification ticket in hexadecimal
+- **is_auth_ticket_for_api**: True if the authentification ticket was generated for an API, false otherwise
+- **create_account**: True if you want that it create a player account for your title
+- **info_request_parameters**: Flags for which pièce of info to return for the user
```
[the remark this answers is not in the document]
Comment (TheDedemon) [reply docs/user/Logins/login-steam.md:52]: Sorry, some French xD

File: docs/user/Logins/login-steam.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+1. Go to your Overview page of your PlayFab Title.
+2. Select the **Add-ons** menu item.
+3. In the list of available **Add-ons**, locate and select Steam
+
+![Login Steam Setup 1](/docs/images/login-steam-setup-1.png)
+
+1. Click on the "Install Steam"
+1. Enter your Steam Application ID
+2. Enter your Steam Web API Key
```
[the remark this answers is not in the document]
Comment (Structed) [reply docs/user/Logins/login-steam.md]: done

File: docs/user/Logins/login-steam-godotsteam.md [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+    steam_auth_ticket.id = auth_ticket
+    steam_auth_ticket.buffer = ticket_buffer
+    steam_auth_ticket.size = ticket_size
+    if result == 1:
+        login(convert_auth_ticket(), true)
+```
+
+## :warning: Troubleshooting
+
+They are many possibles errors, the most commons are:
```
[the remark this answers is not in the document]
Comment (Structed) [reply docs/user/Logins/login-steam-godotsteam.md]: resolved

**Review by Structed — APPROVED** [review]

**Pull request merged**